app/repository/hbase/HBaseData.scala

```diff
@@ -0,0 +1,75 @@
+package repository.hbase
+
+import java.nio.charset.StandardCharsets
+import java.util.Base64
+
+import play.api.libs.json.{ Format, JsResult, JsValue, Json }
+
+/*
+ * HBase only stores bytes, and so all data sent to / read from HBase is Base64 encoded.
+ * We shield the client from this, and provide a format that seamlessly decodes / encodes as required.
+ *
+ * Note that our read API (see RESTRepository) modelled fields as a Map, making the assumption that the
+ * fields in a row would have unique names.  This abstraction unfortunately does not hold for write operations,
+ * where checkAndAct operations require us to construct a json document containing a "row" with two versions of
+ * the same field - the before & after values.  HBaseResponseReader is used for the read API, and uses this
+ * format to parse the HBase response before subsequently adapting it to expose the definition of a Row expected
+ * by that API, rather than the HBaseRow defined here.
+ */
+object HBaseData {
+  case class HBaseRow(key: String, cells: Seq[HBaseCell])
+  case class HBaseCell(column: String, value: String)
+
+  /*
+   * These types model the underlying HBase representation.
+   * As we are relying on automatic format derivation, the field names must match those used by HBase exactly.
+   */
+  private case class EncodedRows(Row: Seq[EncodedRow])
+  private case class EncodedRow(key: String, Cell: Seq[EncodedColumn])
+  private case class EncodedColumn(column: String, `$`: String)
+
+  private implicit val encodedColumnFormat = Json.format[EncodedColumn]
+  private implicit val encodedRowFormat = Json.format[EncodedRow]
+  private implicit val encodedRowsFormat = Json.format[EncodedRows]
+
+  private val Charset = StandardCharsets.UTF_8
+
+  private object Decoder extends ((EncodedRow) => HBaseRow) {
+    def apply(encodedRow: EncodedRow): HBaseRow =
+      HBaseRow(key = decode(encodedRow.key), cells = encodedRow.Cell.map(decodeColumn))
+
+    private def decodeColumn(encodedColumn: EncodedColumn): HBaseCell =
+      HBaseCell(column = decode(encodedColumn.column), value = decode(encodedColumn.`$`))
+
+    private def decode(value: String): String = {
+      val bytes = Base64.getDecoder.decode(value)
+      new String(bytes, Charset)
+    }
+  }
+
+  private object Encoder extends ((HBaseRow) => EncodedRow) {
```

Comment:
```extends (HBaseRow) => EncodedRow``` 
is the literal shorthand for
```extends Function[HBaseRow, EncodedRow]```

As we are talking about an object here, it dictates that the object provides:
```def apply(row: HBaseRow): EncodedRow```

So we are basically saying that our object can be treated as a function.
This has the benefit that it can be transparently used in a map operation:
```rows.map(Encoder)```

Without this, we would need the uglier:
```rows.map(Encoder.apply)```